File: codewars/7kyu/from-the-minimum/dduck-from-the-minimum.py
# https://www.codewars.com/kata/5ac6932b2f317b96980000ca/

from unittest import TestCase

min_value = lambda digits: int("".join(map(str, sorted(list(set(digits))))))

TestCase().assertEqual(min_value([1, 3, 1]), 13)
TestCase().assertEqual(min_value([4, 7, 5, 7]), 457)
TestCase().assertEqual(min_value([4, 8, 1, 4]), 148)

File: codewars/7kyu/from-the-minimum/dduck-from-the-minimum.cpp
// https://www.codewars.com/kata/563b662a59afc2b5120000c6
// $ g++ -std=c++11 space.cpp

#include <iostream>

#include <string>
#include <vector>
#include <set>

using namespace std;

unsigned long long minValue(vector<int> values) {
    std::set<int> set(values.begin(), values.end());
    std::string s;
    for (const auto &v: set) {
        s += std::to_string(v);
    }
    return std::stoi(s);
}

int main() {
    minValue({1, 3, 1});
    return 0;
}
